builtin/sparse-checkout.c

```diff
@@ -230,8 +233,24 @@ static void write_cone_to_file(FILE *fp, struct pattern_list *pl)
 	string_list_clear(&sl, 0);
 
 	hashmap_iter_init(&pl->recursive_hashmap, &iter);
-	while ((entry = hashmap_iter_next(&iter)))
+	while ((entry = hashmap_iter_next(&iter))) {
+		char *slashpos;
+
+		strbuf_setlen(&parent_pattern, 0);
+		strbuf_addstr(&parent_pattern, entry->pattern);
+		slashpos = strrchr(parent_pattern.buf, '/');
+
+		if (slashpos > parent_pattern.buf) {
+			strbuf_setlen(&parent_pattern, slashpos - parent_pattern.buf);
+
+			if (hashmap_contains_path(&pl->recursive_hashmap,
+						  &parent_pattern))
+				continue;
+		}
```

Comment:
Again I'm wondering if this is handling multiple levels deep like if I have
`a` and `a/b/c/d` as my folders?